app/frontend/src/aquifers/components/SingleAquiferMap.vue

```diff
@@ -398,8 +431,56 @@ export default {
         canInteract,
         ecocatLayerIds: [ DATABC_ECOCAT_LAYER_ID ]
       })
+    },
+    updateMapLegendBasedOnVisibleElements() {
+      //gets a list of rendered objects and then updates the legend to only display entries for items that are currently rendered
+      const uniqueRenderedLayerIds = this.getRenderedLayerIds();
+
+      //as cadastrals are a raster layer rather than a vector layer like the others, so seperate logic is required.         
+      this.mapLayers.forEach(layerObj =>{
+        if(uniqueRenderedLayerIds.has(layerObj.id) && layerObj.id != DATABC_CADASTREL_LAYER_ID){
+          this.mapLayers.find((layer) => layer.id === layerObj.id).show = true;
+        } else if(!uniqueRenderedLayerIds.has(layerObj.id) && layerObj.id != DATABC_CADASTREL_LAYER_ID){
+          this.mapLayers.find((layer) => layer.id === layerObj.id).show = false;
```

Comment:
Alternatively, could the function just be shortened down to something like this
```js
      this.mapLayers.forEach(layerObj => {
        this.mapLayers.find((layer) => layer.id === layerObj.id).show = uniqueRenderedLayerIds.has(layerObj.id) && layerObj.id !== DATABC_CADASTREL_LAYER_ID
      })
```

Reply:
for this suggestion, i think it would still change the show value when layerObj.id == DATABC_CADASTREL_LAYER_ID
which is not what we want.
That said, after that runs is when show for cadastral layer is properly set, so it would still work. 

Implemented the stronger equality check, anyways